### ISSUE TITLE
Detecting a complete download of a torrent seeded by libtorrent/libragnar

### ISSUE THREAD

**Issue by ngjermundshaug** [libragnar/ragnar#105]
Hello guys

I know this is a special usecase - but still - here it goes:

We are using libRagnar/libTorrent to distribute file packages to clients. We need to detect when someone has downloaded a filepackage. Our servers are usually the only ones seeding the packages - so the task shouldn't be impossible.

I figure we can do it in several ways:

1. Detect when client downloads the .torrent file. The drawback is that we don't know that the package itself has been downloaded - and clients can distribute the .torrent file to 100 other people that downloads the package - without us detecting it.
2. Detect when a Peer is done downloading - and starts Seeding. Can this be done with libRagnar?
3. Any other alternatives?

**Comment by MozzieMD** [comment]
You need to make a tracker!

**Comment by ngjermundshaug** [comment]
We are running a private instance of OpenTracker. Would like to solve this challenge (as best as it can be done) without having to customize/modify a tracker.

**Comment by MozzieMD** [comment]
In your place i would use PHP-Tracker to make a private tracker, and to populate the database with information! I don't think you can do it other way! Lets hope I'm wrong!

**Comment by vktr** [comment]
@ngjermundshaug I agree with @MozzieMD here, this is what a tracker is meant to do (among other things). Every well-behaving client will send you an announce request as soon as the download is complete.
